contracts/libraries/OracleLibrary.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+
+import '@uniswap/v3-core/contracts/libraries/FullMath.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@uniswap/v3-core/contracts/interfaces/pool/IUniswapV3PoolDerivedState.sol';
+import '@uniswap/v3-core/contracts/libraries/LowGasSafeMath.sol';
+import 'hardhat/console.sol';
+
+library OracleLibrary {
+    function consult(
+        uint32 time,
+        address poolAddress,
+        uint32 startBlockTimestamp,
+        uint32 endBlockTimestamp
+    ) public view returns (uint256 ratioX128) {
```

Comment:
the ratio is a bit hard to work with, in previous examples we had the consult method take an amount of one token and return the amount of the other token, which is great because it lets the caller choose their denominator (instead of using `2**128`)